src/System.IO.Pipelines/tests/StreamPipeWriterTests.cs

```diff
@@ -95,20 +95,24 @@ public async Task WritesUsingGetMemoryWorks()
         [Fact]
         public async Task CanDoMultipleAsyncWritesToStream()
         {
-            var pipe = new Pipe(new PipeOptions(readerScheduler: PipeScheduler.Inline));
+            var pipe = new Pipe();
             PipeWriter writer = PipeWriter.Create(pipe.Writer.AsStream());
+            // This needs to run inline to synchronize the reader and writer
+            TaskCompletionSource<object> waitForRead = null;
 
-            static async Task DoWritesAsync(PipeWriter writer, byte[][] writes)
+            async Task DoWritesAsync(PipeWriter writer, byte[][] writes)
             {
                 for (int i = 0; i < writes.Length; i++)
                 {
+                    waitForRead = new TaskCompletionSource<object>();
                     await writer.WriteAsync(writes[i]);
+                    await waitForRead.Task;
```

Comment:
Add a default timeout here.

Reply:
We don't have timeouts in any of these tests and I don't think we should be adding them just here (they also haven't been a problem since everything is localized to these functions without external resources)